components/net/cookie_storage.rs

```diff
@@ -37,6 +37,23 @@ impl CookieStorage {
         let domain = reg_host(cookie.cookie.domain.as_ref().unwrap_or(&"".to_string()));
         let cookies = self.cookies_map.entry(domain).or_insert(vec![]);
 
+        // https://www.ietf.org/id/draft-ietf-httpbis-cookie-alone-01.txt Step 2
+        if !cookie.cookie.secure && !source.is_secure_protocol() &&
+           cookies.iter().any(|c| {
+                let path_match = if let (&Some(ref new), &Some(ref existing)) = (&cookie.cookie.path, &c.cookie.path) {
+                    Cookie::path_match(new, existing)
+                } else {
+                    cookie.cookie.path == c.cookie.path
+                };
+
+                c.cookie.name == cookie.cookie.name &&
+                c.cookie.secure &&
+                c.cookie.domain == cookie.cookie.domain &&
```

Comment:
Yes, but since it's checking bi-directionally, I don't think it's possible where A == B and B does not domain-match A (or that A doesn not domain-match B).